core/telemetry/src/worker/node.rs

```diff
@@ -116,21 +127,36 @@ where TTrans: Clone + Unpin, TTrans::Dial: Unpin,
 		let mut socket = mem::replace(&mut self.socket, NodeSocket::Poisoned);
 		self.socket = loop {
 			match socket {
-				NodeSocket::Connected(mut conn) =>
+				NodeSocket::Connected(mut conn) => {
 					match NodeSocketConnected::poll(Pin::new(&mut conn), cx, &self.addr) {
-						Poll::Ready(Ok(v)) => match v {}
-						Poll::Pending => break NodeSocket::Connected(conn),
-						Poll::Ready(Err(err)) => {
+						Poll::Ready(Ok(_)) => {
+							break NodeSocket::Connected(conn)
+						},
+						Poll::Pending => {
+							break NodeSocket::Connected(conn)
```

Comment:
```suggestion
```

Reply:
Why this suggestion? We extract the socket then put it back. If you remove this `break` we stay in a `Poisoned` state.

Reply:
I did not looked at the original code close enough.

I wanted to have this `Poll::Ready(Ok(_)) | Poll::Pending => { break Connected() }`.

Reply:
(You can not have multiline suggestions :( )